lib/rpi.pm

```diff
@@ -0,0 +1,36 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+package rpi;
+
+use warnings;
+use strict;
+
+use Exporter 'import';
+use testapi;
+use power_action_utils 'power_action';
+
+our @EXPORT = 'enable_tpm_slb9670';
+
+# Enable TPM on Raspberry Pi
+# https://en.opensuse.org/HCL:Raspberry_Pi3_TPM
+sub enable_tpm_slb9670 {
+    assert_script_run('echo -e "dtparam=spi=on\ndtoverlay=tpm-slb9670" >> /boot/efi/extraconfig.txt');
+    assert_script_run('cat /boot/efi/extraconfig.txt');
+    assert_script_run('echo "tpm_tis_spi" > /etc/modules-load.d/tpm.conf');
+    power_action('reboot', textmode => 1);
+
+    # Add some timeout to wait for reboot
+    sleep(60);
+
+    # Restore SSH connection
+    reset_consoles;
+    select_console('root-ssh');
+
+    assert_script_run('dmesg | grep tpm');
```

Comment:
Maybe you could be more specific than only _tpm_? To ensure this is really the _slb9670_.

Reply:
I copy pasted this code, but indeed this is needed. Because for QEMU and RPI without this setup there is info from IMA which would be also valid for the original grep:

`ima: No TPM chip found, activating TPM-bypass!`